Formula/dps8m.rb

```diff
@@ -1,14 +1,16 @@
 class Dps8m < Formula
-  desc "Simulator for the Multics dps-8/m mainframe"
-  homepage "https://ringzero.wikidot.com/"
-  url "https://gitlab.com/dps8m/dps8m/-/archive/R3.0.0/dps8m-R3.0.0.tar.gz"
-  sha256 "cee3cc0942ceb929d267c4989551526ff50e777706f62bfc449e16d1428a1d2e"
+  desc "Simulator of the 36‑bit GE/Honeywell/Bull 600/6000‑series mainframe computers"
+  homepage "https://dps8m.gitlab.io/"
+  url "https://dps8m.gitlab.io/dps8m-archive/R3.0.0/dps8m-r3.0.0-src.tar.gz"
+  sha256 "e3eac9e4f8b6c7fad498ff1848ba722e1a2e220b793ce02e2ea6a7a585e0c91f"
   license "ICU"
+  revision 1
   head "https://gitlab.com/dps8m/dps8m.git", branch: "master"
 
   livecheck do
-    url :stable
-    regex(/^R?(\d+(?:\.\d+)+)$/i)
+    url "https://dps8m.gitlab.io/dps8m/"
+    regex(/DPS8M R?(\d+(?:\.\d+)+)</i)
+    strategy :page_match
   end
```

Comment:
cc @samford @nandahkrishna

Reply:
For the next DPS8M version (likely R3.0.1) we (the upstream project maintainers) plan to provide a simple plaintext file (at a location such as https://dps8m.gitlab.io/dps8m/VERSION) to facilitate reliable automatic version checking.

Similar to other projects (such as Firefox, OpenBSD, etc.) when we tag a new release in our development repository, it is a strong indication a release is imminent - but it won't be available just yet.

It's the first action which kicks off our CI/CD release process, which builds [literally hundreds of binaries](https://dps8m.gitlab.io/dps8m/Releases/#stable-release)/documentation/cross-references, produces the source kits designed for end-user consumption, rebuilds and deploys our [website](https://dps8m.gitlab.io/), checksums are computed and binaries will be cryptographically signed, the released files are distributed to various mirrors (including [archive.org](https://web.archive.org/web/*/http://dps8m.gitlab.io/*)), etc.  Within a few hours, the released sources and binaries *should* be available for all to download, and the website will then actually be updated, and finally, the official the release announcements made.

So, while it's perfectly acceptable for users to check the development git repos for new tags being pushed, their presence isn't necessarily indicating a new release is available, only that the process (should) have started.

Since most automated version checking is done at reasonable intervals you shouldn't get too many false alarms checking tags in the repos, however, the project website is authoritative.

(Of course, anyone is free to rebuild their own source kits from a git checkout, we don't recommend it, since additional build prerequisites must be installed, extra steps will be needed when building, there will not be any GPG signatures, your checksums won't match those of the official distribution on our website, etc.)

Reply:
As below, we're able to identify the version from the tarball link on the [releases page](https://dps8m.gitlab.io/dps8m/Releases/) and that ensures that livecheck is only surfacing a new version when the related tarball is available.

With that in mind, I'm not sure that the aforementioned `VERSION` URL would be notably beneficial over checking the releases page. It depends on when the `VERSION` information would be updated in your release process. Ideally, it would need to be after the source tarball is available on your website but without a notable delay (such that continuing to check the releases page is more reliable). It would be a much lighter check but part of that is because the `dps8m` website isn't using compression.

The releases page is currently ~246 KB but it would be ~22 KB with Gzip compression or ~17 KB with Brotli. GitLab Pages doesn't compress files on-the-fly but it will serve compressed files if you provide pre-compressed files alongside the uncompressed ones ([related documentation here](https://docs.gitlab.com/ee/user/project/pages/introduction.html#serving-compressed-assets)). 17-22 KB is fine, so there's something to be said for compressing supported files on the website in the interim time.

Reply:
The website would normally be the last thing updated - we deploy with GitLab Pages as the final part of the CI/CD process, so if the file would be updated, the files would be as well.

The motivation for the new file is really more about being able to revamp the website in the future without breaking existing third-party version checks. But until then, I'll ensure that the website won't break the regex.

Thanks for the tip about compression, I'll look into implementing it.